src/models/EnrollUser.js

```diff
@@ -29,15 +30,23 @@ function (Okta, BaseLoginModel) {
         }
       };
       // send createNewAccount flag for new user creation
-      if (this.get('createNewAccount')) {
+      if (this.appState.get('policy').registration.createNewAccount) {
         postData.registration['createNewAccount'] = true;
       }
       return postData;
     },
     getEnrollFormData: function () {
-      return this.manageTransaction(function (transaction) {
-        return transaction.enroll();
-      });
+      if (this.options.appState.get('isUnauthenticated')) {
```

Comment:
I feel that this `if` belongs more in the controller, like you had it before. `model.getEnrollFormData` will always make a request to get that data, which sounds right

Reply:
We dont have to make a call if its not unauthenticated, because the state would beI am using `isUnauthenticated ` to decide if a call needs to be made to fetch the data. If its not unauthenticated then i just resolve using `deferred.resolve(this.appState.get('lastAuthResponse'));`. If I move this check to the controller I feel like I would  have to add a check in the model to see if lastAuthResponse was `PROFILE_REQUIRED`

Reply:
Actually I would have to move stuff into the controller 

``` 
fetchInitialData: function () {
      // If user is unauthenticated and starts enroll flow make a post call to transition state to PROFILE_REQUIRED
      if (this.options.appState.get('isUnauthenticated')) {
        return this.model.getEnrollFormData()
          .then(_.bind(function () {
            this.renderForm();
          }, this));
      } else {
        var deferred = Q.defer();
        deferred.resolve(this.options.appState.get('lastAuthResponse'));
        this.renderForm();
        return deferred.promise;
      }
    },
```
and the model would look like 
``
    getEnrollFormData: function () {
      return this.manageTransaction(function (transaction, setTransaction) {
        return transaction.enroll().then(function (trans) {
          setTransaction(trans);
        });
      });
    },

```
In my approach I was trying to abstract all this logic from the controller.